test/e2e-tests.sh

```diff
@@ -0,0 +1,317 @@
+
+  # Install The eventing-kafka KafkaChannel Implementation
+  ko apply -f "${EVENTING_KAFKA_CONFIG_TEMP_DIR}" || return 1
+
+  # Add The kn-eventing-test-pull-secret (If Present) To ServiceAccount & Restart eventing-kafka Deployment
+  add_kn_eventing_test_pull_secret knative-eventing eventing-kafka-channel-controller eventing-kafka-channel-controller
+
+  # Wait Until All The eventing-kafka Pods Are Running
+  wait_until_pods_running knative-eventing || fail_test "Failed to install the Kafka Channel CRD"
+}
+
+# Uninstall The eventing-kafka KafkaChannel Implementation Via Ko
+function channel_teardown() {
+  echo "Uninstalling KafkaChannel"
+  kubectl delete secret -n knative-eventing kafka-cluster
+  sleep 10 # Give Controller Time To React To Kafka Secret Deletion ; )
+  ko delete --ignore-not-found=true --now --timeout 60s -f "${EVENTING_KAFKA_CONFIG_TEMP_DIR}"
+}
+
+# E2E Test Setup Lifecycle Function
+function test_setup() {
+
+  # Setup Kafka & eventing-kafka Channel
+  kafka_setup || return 1
+  channel_setup || return 1
+
+  # Install kail If Necessary
+  if ! which kail > /dev/null; then
+    bash <( curl -sfL https://raw.githubusercontent.com/boz/kail/master/godownloader.sh) -b "$GOPATH/bin"
+  fi
+
+  # Capture All Logs With Kail
+  kail > ${ARTIFACTS}/k8s.log.txt &
+  local kail_pid=$!
+
+  # Clean Up Kail To Prevent Interference With Shut-Down
+  add_trap "kill $kail_pid || true" EXIT
+
+  # Publish Test Images (Adjust/Revert Ko Paths For Current GOPATH Location In Vendored Eventing Code)
+  echo ">> Publishing test images from eventing"
+  sed -i 's@knative.dev/eventing/test/test_images@knative.dev/eventing-kafka/vendor/knative.dev/eventing/test/test_images@g' "${VENDOR_EVENTING_TEST_IMAGES}"*/*.yaml
+  $(dirname $0)/upload-test-images.sh ${VENDOR_EVENTING_TEST_IMAGES} e2e || fail_test "Error uploading eventing test images"
+  sed -i 's@knative.dev/eventing-kafka/vendor/knative.dev/eventing/test/test_images@knative.dev/eventing/test/test_images@g' "${VENDOR_EVENTING_TEST_IMAGES}"*/*.yaml
+
+  # This upload-test-images.sh command will fail if the test/test_images directory does not exist; uncomment when we have eventing-kafka specific images here
+  # $(dirname $0)/upload-test-images.sh "test/test_images" e2e || fail_test "Error uploading local test images"
+}
+
+# E2E Test Teardown Lifecycle Function
+function test_teardown() {
+
+  # Teardown Kafka & eventing-kafka Channel
+  kafka_teardown
+  channel_teardown
+}
+
+if [[ "$@" =~ "--run-tests" ]]; then
+  # If the local test-created clusterrolebinding is still in place, it needs to be removed or the initialize call will fail.
+  # See https://github.com/knative/test-infra/issues/2446 (remove this kubectl delete command when fix is merged)
+  kubectl delete clusterrolebindings.rbac.authorization.k8s.io "cluster-admin-binding-${USER}" 2> /dev/null
+elif [[ ! -x "$(command -v kubetest2)" ]]; then
```

Comment:
The issue is fixed/merged (just yesterday) - not sure whether we're running against master/ and can rely on it yet or not?

Reply:
The version is tied to a commit in the go.mod, but it was very close to the latest.  I bumped it to today's revision and it seems to work (still needs to install kubetest2 if necessary but no longer fails with a leftover cluster-admin-binding-$USER), so I've removed this particular hack.  Re-ran the tests with a new cluster twice and they seem to work.